### ISSUE TITLE
`no-unused-prop-types`: support `shouldComponentUpdate` nextProps

### ISSUE THREAD

**Issue by xareelee** [yannickcr/eslint-plugin-react#1213]
```js

type LoginMsgContainerProps = {
  getMsgCodeErrorMsg?: string,
};

class LoginMsgContainer extends React.Component {

  props: LoginMsgContainerProps;

  shouldComponentUpdate(props: LoginMsgContainerProps) {
    if (props.getMsgCodeErrorMsg && props.getMsgCodeErrorMsg.length > 0) {
      this.toast.show(props.getMsgCodeErrorMsg);
    }
  }

  ...
}
```

I only use prop `getMsgCodeErrorMsg` in the method `shouldComponentUpdate`, but the linter raises a `react/no-unused-prop-types` error (false positive).

**Comment by ljharb** [comment]
`shouldComponentUpdate` takes `nextProps` - what happens if you name the argument that?

**Comment by xareelee** [comment]
@ljharb changing the param name doesn't help.

The linter reports the error on the line where I declare the type:

```js
type LoginMsgContainerProps = {
  getMsgCodeErrorMsg?: string,   // The linter raises the error on this line and column 24 (react/no-unused-prop-types, ). 
};

  18:24  error  'getMsgCodeErrorMsg' PropType is defined but prop is never used
                                  react/no-unused-prop-types
```

I use the same type for both React component's `props` and the method `shouldComponentUpdate ()`.

**Comment by ljharb** [comment]
If instead of a Flow type, you used standard `.propTypes`, do you get the same error?

**Comment by xareelee** [comment]
@ljharb yes, get the same error with `.propTypes`. I did not find that.

**Comment by ljharb** [comment]
Thanks, it seems like we need to make `no-unused-prop-types` work with `shouldComponentUpdate`.

**Comment by xareelee** [comment]
@ljharb OK, thanks.

Any workaround about this bug for now?

**Comment by ljharb** [comment]
The workaround would be an eslint override comment.

**Comment by ljharb** [comment]
#1232 hopefully has solved this; we can reopen after the next release if not.
